DigitalLearningSolutions.Data/Services/CourseContentService.cs

```diff
@@ -80,18 +80,55 @@ GROUP BY
             ).FirstOrDefault();
         }
 
-        public int GetProgressId(int candidateId, int customisationId)
+        public int? GetOrCreateProgressId(int candidateId, int customisationId, int centreId)
         {
-            // TODO HEEDLS-202: change QueryFirstOrDefault to creating progress record if not found
-            return connection.QueryFirstOrDefault<int>(
-                @"SELECT ProgressId
-                        FROM Progress
-                        WHERE CandidateID = @candidateId
-                          AND CustomisationID = @customisationId",
-                new { candidateId, customisationId }
+            if (DoesProgressExist(candidateId, customisationId))
+            {
+                return GetProgressId(candidateId, customisationId);
+            }
+
+            var errorCode = connection.QueryFirst<int>(

```

Comment:
Do you need `QueryFirst`? Would just `Query` work? 🙂

Reply:
Query returns IEnumerable<int> even though it's guaranteed to only return one value.

Reply:
Ah fair enough, this is fine then 🙂